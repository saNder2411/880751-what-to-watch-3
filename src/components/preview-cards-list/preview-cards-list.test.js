import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import WithPreviewCardsListState from '../../hocs/with-preview-cards-list-state/with-preview-cards-list-state.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.jsx';

const mockStore = configureStore();

const store = mockStore({
  genre: `All Genre`,
  cardsData: [],
  promoCardData: {},
  reviews: [],
  newReviews: [],
});

const MockWrappedPreviewCardsList = withActiveItem(WithPreviewCardsListState);

describe(`Render PreviewCardsList`, () => {
  it(`Should PreviewCardsList render correctly in Main`, () => {
    const markup = renderer
      .create(
          <Provider store={store} >
            <BrowserRouter>
              <Switch>
                <Route
                  path='/'
                  exact
                  render={() => <MockWrappedPreviewCardsList />}
                />
              </Switch>
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });

  it(`Should PreviewCardsList render correctly in CardScreen`, () => {
    const markup = renderer
      .create(
          <Provider store={store} >
            <BrowserRouter>
              <Switch>
                <Route
                  path='/cards:id'
                  exact
                  render={() => <MockWrappedPreviewCardsList selectedCardId={0}/>}
                />
              </Switch>
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });
});
