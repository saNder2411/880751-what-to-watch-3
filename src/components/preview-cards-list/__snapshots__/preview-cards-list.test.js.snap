// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PreviewCardsList Should PreviewCardsList render correctly in CardScreen 1`] = `null`;

exports[`Render PreviewCardsList Should PreviewCardsList render correctly in Main 1`] = `
<div
  className="catalog__movies-list"
/>
`;
